example/src/App.vue

```diff
@@ -296,43 +300,60 @@ export default defineComponent({
     onChangeEditData() {
       console.log('on change data');
     },
-    registerJsonSchemaValidation(editor: EditorApi) {
-      const modelUri = monaco.Uri.parse(
-        'json://core/specification/schema.json'
-      );
-      configureJsonSchemaValidation(editor, modelUri);
-    },
-    registerUISchemaValidation(editor: EditorApi) {
-      const modelUri = monaco.Uri.parse(
-        'json://core/specification/uischema.json'
-      );
-      configureUISchemaValidation(editor, modelUri);
-    },
-    registerDataValidaton(editor: EditorApi) {
-      const example = this.example;
-      if (example && example.schema) {
-        if (!Object.prototype.hasOwnProperty.call(example.schema, '$schema')) {
-          example.schema['$schema'] = jsonSchemaDraft7.uri;
-        }
-        if (!Object.prototype.hasOwnProperty.call(example.schema, '$id')) {
-          (example.schema as any)['$id'] = example.schemaId;
-        }
+    registerValidatons(editor: EditorApi) {
+      // register validation schemas
+      for (let [index, example] of examples.entries()) {
+        configureJsonSchemaValidation(
+          editor,
+          monaco.Uri.parse(`${index}-schema.json`)
+        );
+        configureUISchemaValidation(
+          editor,
+          monaco.Uri.parse(`${index}-uischema.json`)
+        );
+
+        const schema = {
+          ...example.input.schema,
+          title: example.title,
+        };
 
-        configureDataValidation(editor, {
-          uri: (example.schema as any)['$id'],
-          schema: example.schema,
-        });
+        if (example && example.input.schema) {
+          configureDataValidation(
+            editor,
+            `inmemory://${index}/schema.json`,
+            monaco.Uri.parse(`${index}-data.json`),
+            schema
+          );
+        }
       }
     },
   },
   computed: {
     example(): JsonInput {
       const e = this.examples[this.selectedExample.value];
       if (e) {
+        let input = (e.input as JsonInput)!;
+        if (input.schema !== undefined && input?.resolvedSchema === undefined) {
+          // have the original schema while resolving the actual schema
+          input.resolvedSchema = input.schema;
+
+          $RefParser.dereference(
```

Comment:
I would like to strongly suggest to use [JsonRefs](https://github.com/whitlockjc/json-refs) instead of the json-schema-ref-parser. For example the json schema ref parser modifies the schema it is handed over to resolve.